topic-operator/src/main/java/io/strimzi/operator/topic/TopicStoreTopologyProvider.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic;
+
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.ForeachAction;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.Stores;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.Supplier;
+
+/**
+ * Kafka Streams topology provider for TopicStore.
+ */
+public class TopicStoreTopologyProvider implements Supplier<Topology> {
+    private final String storeTopic;
+    private final String topicStoreName;
+    private final Properties kafkaProperties;
+    private final ForeachAction<? super String, ? super Integer> dispatcher;
+
+    public TopicStoreTopologyProvider(
+            String storeTopic,
+            String topicStoreName,
+            Properties kafkaProperties,
+            ForeachAction<? super String, ? super Integer> dispatcher
+    ) {
+        this.storeTopic = storeTopic;
+        this.topicStoreName = topicStoreName;
+        this.kafkaProperties = kafkaProperties;
+        this.dispatcher = dispatcher;
+    }
+
+    @Override
+    public Topology get() {
+        StreamsBuilder builder = new StreamsBuilder();
+
+        // Simple defaults
+        Map<String, String> configuration = new HashMap<>();
+        configuration.put(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT);
+        configuration.put(TopicConfig.MIN_COMPACTION_LAG_MS_CONFIG, "0");
+        configuration.put(TopicConfig.SEGMENT_BYTES_CONFIG, String.valueOf(64 * 1024 * 1024));
+
+        // Input topic command -- store topic
+        // Key is topic name -- which is also used for KeyValue store key
```

Comment:
Since TO has to deal with topic names being represented as Kube resource names, it's always best to be explicit about which name you mean.

```suggestion
        // Key is Kafka topic name -- which is also used for KeyValue store key
```